src/layouts/PLayoutResizable/PLayoutResizable.vue

```diff
@@ -0,0 +1,174 @@
+<template>
+  <section ref="container" class="p-layout-resizable">
+    <aside v-if="slots.aside" ref="aside" class="p-layout-resizable__aside" :class="classes.aside">
+      <slot name="aside" v-bind="{ collapsed, expand, collapse, toggle }" />
+    </aside>
+
+    <div
+      v-if="slots.default && slots.aside"
+      class="p-layout-resizable__divider"
+      :class="classes.divider"
+      @mousedown="start"
+      @mouseup="handleAsideClick"
+    />
+
+    <main v-if="slots.default" class="p-layout-resizable__main" :class="classes.main">
+      <slot v-bind="{ collapsed, expand, collapse, toggle }" />
+    </main>
+  </section>
+</template>
+
+<script lang="ts" setup>
+  import { computed, ref, useSlots } from 'vue'
+
+  const props = defineProps<{
+    collapsePoint?: number,
+    disabled?: boolean,
+  }>()
+
+  const slots = useSlots()
+
+  const container = ref<HTMLDivElement>()
+  const aside = ref<HTMLDivElement>()
+
+  const dragging = ref(false)
+  const collapsed = ref(false)
+
+  const classes = computed(() => ({
+    aside: {
+      'p-layout-resizable--events-none': dragging.value,
+      'p-layout-resizable__aside--collapsed': collapsed.value,
+    },
+    divider: {
+      'p-layout-resizable__divider--resizable': !props.disabled && !dragging.value,
+      'p-layout-resizable__divider--collapsed': collapsed.value,
+    },
+    main: {
+      'p-layout-resizable--events-none': dragging.value,
+    },
+  }))
+
+  let asideLeft = 0
+  const drag = (event: MouseEvent): void => {
+    if (props.disabled || !container.value || !aside.value) {
+      return
+    }
+
+    if (props.collapsePoint) {
+      collapsed.value = event.clientX - asideLeft < props.collapsePoint
+    }
+
+    aside.value.style.width = `${event.clientX - asideLeft}px`
+  }
+
+  const start = (): void => {
+    if (props.disabled || !container.value || !aside.value) {
+      return
+    }
+
+    const asideBounds = aside.value.getBoundingClientRect()
+    asideLeft = asideBounds.left
+    document.body.style.cursor = 'col-resize'
```

Comment:
manipulating this directly seems a little sketchy. Can we make a class and toggle the class on the body?

Reply:
Lol you're completely right, idk why I went with that - added a class method instead.